pkg/docker/docker.go

```diff
@@ -126,6 +126,62 @@ type RunContainerOptions struct {
 	CapDrop          []string
 }
 
+// asDockerConfig converts a RunContainerOptions into a Config understood by the
+// docker client library.
+func (rco RunContainerOptions) asDockerConfig() docker.Config {
+	return docker.Config{
+		Image: getImageName(rco.Image),
+		User:  rco.User,
+		// TODO why do we call runContainerTar with a "partial"
+		// docker.Config?
+
+		Env: rco.Env,
+
+		// TODO review how callers use these options to pass stdin.
+		// Probably they are not necessary, as they're related to
+		// attaching to the container (that is often unnecessary).
+		OpenStdin: rco.Stdin != nil,
+		StdinOnce: rco.Stdin != nil,
+		// TODO surprinsingly, we do not set AttachStdin...
+
+		AttachStdout: rco.Stdout != nil,
+	}
+}
+
+// asDockerHostConfig converts a RunContainerOptions into a HostConfig
+// understood by the docker client library.
+func (rco RunContainerOptions) asDockerHostConfig() docker.HostConfig {
+	hostConfig := docker.HostConfig{
+		CapDrop: rco.CapDrop,
+
+		PublishAllPorts: rco.TargetImage,
+		// TODO why didn't we always set NetworkMode?!
```

Comment:
What does the `config.TargetImage` boolean mean?

Prior code would only set `NetworkMode` if `TargetImage` was true, or if `NetworkMode` is not the empty string (redundant check). If `TargetImage` was false and `NetworkMode` non-empty, it would be set as well... so seems that "we always set it" anyway, but hidden under conditionals...

Still, what does `TargetImage` means?